src/session_server.c

```diff
@@ -4278,3 +4430,83 @@ nc_server_is_mod_ignored(const char *mod_name)
 
     return ignored;
 }
+
+API int
+nc_server_set_unix_socket_path(const char *endpoint_name, const char *socket_path)
+{
+    int rc = 0;
+    LY_ARRAY_COUNT_TYPE i;
+    struct nc_server_unix_path_entry *pentry = NULL;
+
+    NC_CHECK_ARG_RET(NULL, endpoint_name, socket_path, 1);
+
+    /* CONFIG WRITE LOCK */
+    pthread_rwlock_wrlock(&server_opts.config_lock);
+
+    /* try to see if the path for this endpoint already exists */
+    LY_ARRAY_FOR(server_opts.unix_paths, i) {
+        if (!strcmp(server_opts.unix_paths[i].endpt_name, endpoint_name)) {
+            pentry = &server_opts.unix_paths[i];
+            break;
+        }
+    }
+    if (!pentry) {
+        /* create a new entry */
+        LY_ARRAY_NEW_GOTO(NULL, server_opts.unix_paths, pentry, rc, cleanup);
+        pentry->endpt_name = strdup(endpoint_name);
+        NC_CHECK_ERRMEM_GOTO(!pentry->endpt_name, rc = 1, cleanup);
+    } else {
+        /* free the old path */
+        free(pentry->path);
+    }
+
+    pentry->path = strdup(socket_path);
+    NC_CHECK_ERRMEM_GOTO(!pentry->path, rc = 1, cleanup);
+
+cleanup:
+    /* CONFIG WRITE UNLOCK */
+    pthread_rwlock_unlock(&server_opts.config_lock);
+    return rc;
+}
+
+API const char *
+nc_server_get_unix_socket_path(const char *endpoint_name)
+{
+    const char *socket_path = NULL;
+    LY_ARRAY_COUNT_TYPE i;
+
+    NC_CHECK_ARG_RET(NULL, endpoint_name, NULL);
+
+    /* CONFIG READ LOCK */
+    pthread_rwlock_rdlock(&server_opts.config_lock);
+
+    /* try to find the path for this endpoint */
+    LY_ARRAY_FOR(server_opts.unix_paths, i) {
+        if (!strcmp(server_opts.unix_paths[i].endpt_name, endpoint_name)) {
+            socket_path = server_opts.unix_paths[i].path;
+            break;
+        }
+    }
+
+    /* CONFIG READ UNLOCK */
+    pthread_rwlock_unlock(&server_opts.config_lock);
+    return socket_path;
```

Comment:
Not safe to return, it can later be changed. Must be dynamically allocated it seems.